api/envoy/extensions/filters/udp/dns_filter/v3alpha/dns_filter.proto

```diff
@@ -44,9 +46,32 @@ message DnsFilterConfig {
     }
   }
 
+  // This message contains the configuration for the Dns Filter operating
+  // in a client context. This message will contain the timeouts, retry,
+  // and forwarding configuration for Envoy to make DNS requests to other
+  // resolvers
+  message ClientContextConfig {
+    option (udpa.annotations.versioning).previous_message_type =
+        "envoy.config.filter.udp.dns_filter.v2alpha.DnsFilterConfig.ClientContextConfig";
+
+    // Sets the maximum time we will wait for the upstream query to complete
+    // We allow 5 ms for the upstream resolution to complete, so the minimum
+    // value here is 5
+    google.protobuf.Duration resolver_timeout = 1 [(validate.rules).duration = {gt {}}];
+
+    // A list of DNS servers to which we can forward queries
+    repeated string upstream_resolvers = 2 [(validate.rules).repeated = {
+      min_items: 1
+      items {string {min_len: 3}}
+    }];
+  }
+
   // The stat prefix used when emitting DNS filter statistics
   string stat_prefix = 1 [(validate.rules).string = {min_len: 1}];
 
   // Server context configuration
   ServerContextConfig server_config = 2;
+
+  // Client context configuration
```

Comment:
Can you clarify what happens if this is absent?

Reply:
If no client context is specified, then envoy will not forward queries to external resolvers.   In this case, if the domain name is not specified in the static configuration, Envoy will immediately respond to the query with a NOANSWER response code.

There is a use case where Envoy could operate as a DNS Relay so a minimal client context is needed with no resolvers and just the timeout.  In this case, all queries will be forwarded to the system configured DNS.

Reply:
That makes sense. Please spell it out in the docs.

Reply:
Ping on this.

Reply:
I updated the .rst.   Later I added a comment in the protobuf.  Let me know if more is needed.